### PR TITLE
users: Fix creation of locked accounts

### PR DESCRIPTION
**Pull request by mvollmer**
Running "chpasswd" last will undo the earlier "usermod --lock", so we
do it the other way around.

The added checks in the test also remove a race condition. We need to
wait for the unlocking finish before changing the groups.

### CODE REVIEW

**Comment by stefwalter** [comment]
This looks like it introduces a real failure:

https://fedorapeople.org/groups/cockpit/logs/pull-10327-20181017-074545-7bbefe59-verify-debian-testing/log.html

**Comment by mvollmer** [comment]
> This looks like it introduces a real failure:

Indeed.  The screen scraping of `passwd -S` didn't work on Debian.

**Comment by martinpitt** [comment]
There are still several check-accounts test failures here which are new.

File: test/verify/check-accounts [hunk 1]
```diff
@@ -23,6 +23,7 @@ import datetime
 import parent
 from testlib import *
 
+@skipImage("Don't run upstream tests against distro packages", "rhel-7-5-distropkg", "rhel-7-6-distropkg")
```
Comment (martinpitt) [inline test/verify/check-accounts]: Seems okay as per our IRC discussion. But then please remove the existing two -distropkg special cases from the test below.

Comment (mvollmer) [reply test/verify/check-accounts]: Of course! Fixed.

**Pull request merged**